docs/fassets/developer-guides/12-fassets-autoredeem.mdx

```diff
@@ -0,0 +1,580 @@
+    "0x14bfb521e318fc3d5e92A8462C65079BC7d4284c",
+  COSTON2_COMPOSER: process.env.COSTON2_COMPOSER || "",
+  COSTON2_EID: EndpointId.FLARE_V2_TESTNET,
+  EXECUTOR_GAS: 400_000, // Gas for receiving on Coston2
+  COMPOSE_GAS: 700_000, // Gas for compose execution
+  SEND_AMOUNT: "10", // 10 FXRP = 1 lot
+  XRP_ADDRESS: "rpHuw4bKSjonKRrKKVYUZYYVedg1jyPrmp", // Your XRP address
+};
+```
+
+### How to Run
+
+**PREREQUISITE: You must have FXRP tokens on Hyperliquid EVM Testnet before running this script.**
+
+If you don't have FXRP on Hyperliquid yet:
+
+- Run `bridgeToHyperEVM.ts` first (Step 1 - see above section)
+- Wait for the bridge transaction to complete (2-5 minutes)
+- Verify your FXRP balance on Hyperliquid EVM before proceeding
+
+Once you have FXRP on Hyperliquid:
+
+1. **Deploy FAssetRedeemComposer** (first time only):
+
+   ```bash
+   npx hardhat deploy --network coston2 --tags FAssetRedeemComposer
+   ```
+
+2. **Configure Environment**:
+
+   ```bash
+   # .env file
+   HYPERLIQUID_TESTNET_RPC_URL=https://api.hyperliquid-testnet.xyz/evm
+   DEPLOYER_PRIVATE_KEY=your_private_key_here
+   COSTON2_COMPOSER=0x... # Required! Set this after deploying the composer
+   HYPERLIQUID_FXRP_OFT=0x14bfb521e318fc3d5e92A8462C65079BC7d4284c
+   ```
+
+3. **Update XRP Address**:
+   - Edit `CONFIG.XRP_ADDRESS` in the script to your XRP ledger address
+   - This is where you'll receive the native XRP
+   - Must be a valid XRP address format (starts with 'r')
+
+4. **Run the Script on Hyperliquid Testnet**:
+   ```bash
+   yarn hardhat run scripts/fassets/autoRedeemHyperEVM.ts --network hyperliquidTestnet
+   ```
+
+### Expected Output
+
+```
+Using account: 0x742d35Cc6634C0532925a3b844Bc454e4438f44e
+✓ Composer configured: 0x123...
+
+📋 Redemption Parameters:
+Amount: 10.0 FXRP
+XRP Address: rpHuw4bKSjonKRrKKVYUZYYVedg1jyPrmp
+Redeemer: 0x742d35Cc6634C0532925a3b844Bc454e4438f44e
+
+Connecting to FXRP OFT on Hyperliquid EVM: 0x14bfb521e318fc3d5e92A8462C65079BC7d4284c
+
+✓ Connected to FXRP OFT: 0x14bfb521e318fc3d5e92A8462C65079BC7d4284c
+OFT address: 0x14bfb521e318fc3d5e92A8462C65079BC7d4284c
+
+Compose message encoded
+
+💰 Current FXRP balance: 25.0
+Sufficient balance
+
+💵 LayerZero Fee: 0.002456 HYPE
+
+🚀 Sending 10.0 FXRP to Coston2 with auto-redeem...
+Target composer: 0x123...
+Underlying address: rpHuw4bKSjonKRrKKVYUZYYVedg1jyPrmp
+
+✓ Transaction sent: 0xdef456...
+Waiting for confirmation...
+✅ Confirmed in block: 9876543
+
+🎉 Success! Your FXRP is on the way to Coston2!
+
+📊 Track your cross-chain transaction:
+https://testnet.layerzeroscan.com/tx/0xdef456...
+
+⏳ The auto-redeem will execute once the message arrives on Coston2.
+XRP will be sent to: rpHuw4bKSjonKRrKKVYUZYYVedg1jyPrmp
+```
+
+<details>
+<summary>View `autoRedeemHyperEVM.ts` source code</summary>
+
+<CodeBlock language="typescript" title="scripts/fassets/autoRedeemHyperEVM.ts">
+  {autoRedeemHyperEVM}
+</CodeBlock>
+
+</details>
+
+## FAQ
```

Comment:
As this guide is quite long I would avoid such section, it can be scattered within the guide itself.